pym/gentoolkit/eclean/search.py

```diff
@@ -656,15 +657,33 @@ def findPackages(
             binpkg_metadata = dict(zip(keys, bin_dbapi.aux_get(cpv, keys)))
             ebuild_metadata = dict(zip(keys, port_dbapi.aux_get(cpv, keys)))
 
-            if _deps_equal(
-                " ".join(binpkg_metadata[key] for key in dep_keys),
-                binpkg_metadata["EAPI"],
-                " ".join(ebuild_metadata[key] for key in dep_keys),
-                ebuild_metadata["EAPI"],
-                libc_deps,
-                frozenset(binpkg_metadata["USE"].split()),
-            ):
-                continue
+            deps_binpkg = " ".join(binpkg_metadata[key] for key in dep_keys)
+            deps_ebuild = " ".join(ebuild_metadata[key] for key in dep_keys)
+            try:
+                if _deps_equal(
+                    deps_binpkg,
+                    binpkg_metadata["EAPI"],
+                    deps_ebuild,
+                    ebuild_metadata["EAPI"],
+                    libc_deps,
+                    frozenset(binpkg_metadata["USE"].split()),
+                ):
+                    continue
+            except InvalidDependString as er:
+                print(
+                    pp.error("findPackages", "InvalidDependString found for: %s" % cpv),
+                    file=sys.stderr,
+                )
+                print(
+                    pp.error("findPackages", "deps_binpkg: %s" % deps_binpkg),
+                    file=sys.stderr,
+                )
+                print(
+                    pp.error("findPackages", "deps_ebuild: %s" % deps_ebuild),
+                    file=sys.stderr,
+                )
+                print(pp.error(er))
+                exit(1)
```

Comment:
I suppose if the binpkg was at fault here then we might behave as though _deps_equal returned False and just delete binpkg with possibly a warning message. Invalid ebuild deps being fatal seems reasonable though.

Reply:
Yeah, a bad binpkg should get deleted.  I figured this would be a good start to at least find out what was broken on his 2 systems.   We can leave this PR open for awhile until we here back from the Stuart.

Plus I need to create some errors to properly test it.